pkg/runtime/reconciler.go

```diff
@@ -60,6 +63,12 @@ func (r *reconciler) BindControllerManager(mgr ctrlrt.Manager) error {
 	if r.rmf == nil {
 		return ackerr.NilResourceManagerFactory
 	}
+	clusterConfig := mgr.GetConfig()
+	kc, err := kubernetes.NewForConfig(clusterConfig)
+	if err != nil {
+		return err
+	}
+	r.cache = ackrtcache.New(kc, r.log)
 	r.kc = mgr.GetClient()
```

Comment:
Any reason why you did the above instead of just:

```go
r.kc = mgr.GetClient()
r.cache = ackrtcache.New(r.kc, r.log)
```

Reply:
my bad for using `kc` as a variable name ... `clientset` makes much more sense. 
`ackrtcache.New` first parameter takes a `client-go/kubernetes.Interface` , while `mgr.GetClient` returns a `client.Client` (from `sigs.k8s.io/controller-runtime/pkg/client`)

Reply:
Changed `kc` to `clientset`

Reply:
Sorry, you misunderstood me :) I was saying that the clientset object is already being created on line 71 with:

```go
r.kc = mgr.GetClient()
```

so you don't have to create another one with the call to `mgr.GetConfig() and then `kubernetes.NewForConfig()`. You can just pass the reconciler's `kc` attribute (which is the kubernetes clientset) to your `ackrtcache.New()` function.

Reply:
Ahhh i see now... i think i tried this approach before but something wasn't compiling

Reply:
i think it's because: `client-go/kubernetes.NewForConfig` returns a `kubernetes.Interface`, `mgr.Client` return a `ctrlrt/client.Client` and `client.Client` doesn't implement the `kubernetes.Interface` interface

Reply:
hmmm. interesting... OK, soon as I'm done with a customer call I'll look into perhaps a cleaner way to do this.

Reply:
Just closing this out, but I've asked @A-Hilaly to create an issue with a TODO about this and we're going to merge this as-is.